src/systems/model_parsing.jl

```diff
@@ -74,9 +75,9 @@ function _model_macro(mod, fullname::Union{Expr, Symbol}, expr, isconnector)
     push!(exprs.args, :(parameters = []))
     # We build `System` by default; vectors can't be created for `System` as it is
     # a function.
-    push!(exprs.args, :(systems = ModelingToolkit.AbstractSystem[]))
-    push!(exprs.args, :(equations = Union{Equation, Vector{Equation}}[]))
-    push!(exprs.args, :(defaults = Dict{Num, Union{Number, Symbol, Function}}()))
+    push!(exprs.args, :(systems = $MTK.AbstractSystem[]))
```

Comment:
Is prepending `$MTK.` to everything really the solution? What if we did this instead:
```suggestion
    push!(exprs.args, :(systems = $AbstractSystem[]))
```

Also wouldn't this need to be done in more places? e.g. where `@variables` is called.

Reply:
See the discussion around https://discourse.julialang.org/t/adding-method-to-function-in-macro/128613/6 . If you're going to `esc` the expression, then you need either `$MTK.AbstractSystem` or `$AbstractSystem`, but there are places where the latter will fail. In particular, you can't `$@some_macro` obviously. I always use the former style for that reason, but I don't mind changing this PR to `$AbstractSystem` if that's your preference.

> Also wouldn't this need to be done in more places? 

Yes, absolutely. I didn't want to venture outside of my MTK comfort zone.

> e.g. where @variables is called.

?

Reply:
Yes, I always prefix functions, types and macros with `$MyModule.` when using `esc` It's the flip side of skipping macro hygiene.

In this particular case it doesn't matter because the user will feel compelled to `import ModelingToolkit: @variables` anyway, but that's not reliable in general.